spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDParser.java

```diff
@@ -43,4 +45,29 @@
 	 *         described by the DSL.
 	 */
 	List<ModuleDescriptor> parse(String name, String config, ParsingContext type);
+
+	public static class DefinitionOrException {
```

Comment:
What does this have to do with transport changes? Perhaps the transient repository should not exist either (isn't it just a map of what's already been retrieved to avoid hitting the real repository again?). The reason I'm concerned is that while you are making these changes, we are also trying to simplify the parser where far too much responsibility has accumulated. At the moment, the changes you are pushing down to the parser appear to be only relevant for the use-case that hits FloController, so it feels like a responsibility leak in the other direction.

In any case, the classes with OR in the name are a red flag to me. Either in the parser or the controller, if such classes remain, I'd prefer to see something like ParseResult with hasErrors(), getErrors(), etc.